03/app.js

```diff
@@ -1,27 +1,65 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import { v4 as uuid } from 'uuid';
 
 import Category from './Category';
 import Cart from './Cart';
+import Product from './Product';
 
 import data from './data.json';
 
 class App extends React.Component {
-    state = {
-        cart: [],
-    }
-    
-    render() {
-        return (
-            <section>
-                <Category />
-                <Cart />
-            </section>
-        )
-    }
+  state = {
+    cart: [],
+  };
+
+  addToTheCart = (product) => {
+    const cartCopy = [...this.state.cart];
+    product.isCart = true;
+
+    cartCopy.push(product);
+    this.setState({ cart: cartCopy });
+  };
+
+  removeFromTheCart = (product) => {
+    const cartCopy = [...this.state.cart];
+    product.isCart = false;
+
+    const productIndex = cartCopy.indexOf(product);
+    cartCopy.splice(productIndex, 1);
```

Comment:
Skoro usuwasz ten element to po co mu ustawiać `isCart`?
Tutaj prościej byłoby użyć `filter()`

Reply:
Zrobiłem to w celu wyrenderowania buttona w komponencie <Product />